src/TicketMap/index.tsx

```diff
@@ -533,21 +513,28 @@ export default class TicketMap extends Component<Props & DefaultProps, State> {
     return (
       <div
         ref={element => { this.rootRef = element as HTMLElement }}
-        onMouseOver={this.onMouseOver}
-        onMouseOut={this.onMouseOut}
-        onMouseMove={this.onMouseMove}
-        onClick={this.onClick}
-        style={{
-          position: 'relative',
-          fontFamily: this.props.mapFontFamily
+        onMouseOver={({ clientX, clientY, target }: React.MouseEvent<HTMLElement>) => this.doHover(clientX, clientY, target as HTMLElement)}
+        onTouchStart={({ touches, target }: React.TouchEvent<HTMLElement>) => this.doHover(touches[0].clientX, touches[0].clientY, target as HTMLElement)}
+        onMouseOut={({ relatedTarget }: React.MouseEvent<HTMLElement>) => this.doHoverCleanup(relatedTarget as HTMLElement)}
+        onMouseMove={({ nativeEvent }: React.MouseEvent<HTMLElement>) => this.setState({
+          tooltipX: nativeEvent.offsetX,
+          tooltipY: nativeEvent.offsetY
+        })}
+        onClick={this.selectSectionOrZone}
+        onTouchMove={({ target }: React.TouchEvent<HTMLElement>) => {
+          this.doHoverCleanup(target as HTMLElement)
+          this.setState({ isDragging: true })
```

Comment:
This is really difficult to parse at a glance, especially on thinner screens. It also significantly breaks the flow of reading jsx.

Also, these handlers are re-created each time render is called. This will definitely widen the component's memory footprint and perhaps cause the child element to spend more time than necessary responding to the "updated" handler props. We know the value of these handlers when a TicketMap is created, so why not generate them at construction time and call it a day?